### PR TITLE
Servlet/authentication

### PR DESCRIPTION
**Pull request by AndrzejPodo**
Adds:
-servlet for obtaining User details
-servlet for obtaining Login Url
-servlet for obtaining Logout Url

### CODE REVIEW

**Comment by PaulKovalov** [comment]
@googlebot I consent.

File: sticknotes-frontend/src/app/login-page/login-page.component.ts [hunk 1]
```diff
@@ -7,9 +9,19 @@ import { Component, OnInit } from '@angular/core';
...
+    })
+  }
 
   ngOnInit(): void {
   }
 
+  goToUrl(): void {
+    console.log(this.loginUrl);
+    this.document.location.href = this.loginUrl;
+}
```
Comment (PaulKovalov) [inline sticknotes-frontend/src/app/login-page/login-page.component.ts:26]: broken indent

File: sticknotes-frontend/src/app/login-page/login-page.component.ts [hunk 2]
```diff
@@ -7,9 +9,19 @@ import { Component, OnInit } from '@angular/core';
...
+    this.http.get(`api/login-url/`, {responseType: 'text'}).subscribe((loginUrl: string) => {
+      this.loginUrl = loginUrl;
+    })
+  }
 
   ngOnInit(): void {
   }
 
+  goToUrl(): void {
+    console.log(this.loginUrl);
```
Comment (PaulKovalov) [inline sticknotes-frontend/src/app/login-page/login-page.component.ts:24]: debug log

File: sticknotes-frontend/src/app/services/user.service.ts [hunk 3]
```diff
@@ -10,28 +12,42 @@ export class UserService {
...
+  constructor(private http: HttpClient) {
+    this.fetch();
+  }
+
+  private fetch(): Observable<User> {
+    return this.http.get('api/user/').pipe(map((fetchedUser: User) => {
       this.authenticated.next(true);
       this.userSubject.next(fetchedUser);
-    });
+      return this.userSubject.value;
```
Comment (PaulKovalov) [inline sticknotes-frontend/src/app/services/user.service.ts:23]: why here you return `.value`, and in `catchError` clause you return `.asObservable()`?

**Review by PaulKovalov — CHANGES_REQUESTED** [review]

File: sticknotes-frontend/src/app/services/user.service.ts (hunk 3 above)
Comment (AndrzejPodo) [reply sticknotes-frontend/src/app/services/user.service.ts:23]: because it's the expected way
image
but I figured out catching error here is not needed.

**Review by PaulKovalov — APPROVED** [review]

**Pull request merged**